std/traits.d

```diff
@@ -8402,13 +8405,18 @@ private template getSymbolsByUDAImpl(alias symbol, alias attribute, names...)
 */
 template allSameType(T...)
 {
-    static if (T.length <= 1)
+    static foreach (idx, Ti; T)
     {
-        enum bool allSameType = true;
+        static if (idx + 1 < T.length &&
+                   !is(typeof(allSameType) == bool) &&
+                   !is(T[idx] == T[idx + 1]))
+        {
+            enum bool allSameType = false;
+        }
     }
-    else
+    static if (!is(typeof(allSameType) == bool))
     {
-        enum bool allSameType = is(T[0] == T[1]) && allSameType!(T[1..$]);
+        enum bool allSameType = true;
```

Comment:
Hmm looks like DScanner's check is bogus here:

```
std/traits.d(8419:19)[warn]: Variable "allSameType" has the same name as a variable defined on line 8414.
```

I don't know why the other very similar code is accepted though.
However, you can ignore this module for this check by adding the following to the .dscanner.ini

```
label_var_same_name_check="-std.traits"
```

(Ideally preceded by a reference to a DScanner issue)

Reply:
Done.